feat(curriculum): add password generator project to Python curriculum (#52444)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/20-upcoming-python/learn-regular-expressions-by-building-a-password-generator/6577333feab1e8e927014f03.md

---
id: 6577333feab1e8e927014f03
title: Step 63
challengeType: 20
dashedName: step-63
---

# --description--

> Having `all([expression for i in iterable])`, means that a new list is created by evaluating `expression` for each `i` in `iterable`. After the `all()` function iterates over the newly created list, the list is deleted, since is no longer needed.

Reviewer: ```suggestion
Having `all([expression for i in iterable])`, means that a new list is created by evaluating `expression` for each `i` in `iterable`. After the `all()` function iterates over the newly created list, the list is deleted automatically, since it is no longer needed.
```

> Memory can be saved by using a generator expression. Generators expressions follow the same syntax of the list comprehensions but they require parentheses instead of square backets.

Reviewer: ```suggestion
Memory can be saved by using a generator expression. Generator expressions follow the syntax of list comprehensions but they use parentheses instead of square brackets.
```

Change your list comprehension into a generator expression by removing the square brackets.

# --hints--

> You should turn your list comprehension into a generator expression by removing the square brackets from it.

Reviewer: ```suggestion
You should turn your list comprehension into a generator expression by removing the square brackets.
```
